### PR TITLE
Add amber

### PR DESCRIPTION
**Pull request by bdesham**
###### Motivation for this change

This PR adds Amber, a search-and-replace tool for code.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build amber

File: pkgs/tools/text/amber/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "dalance";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0jwrkd6qhxj2mqsfmhk687k15f7gf36gjyxnynj0yh8db2db6mjc";
+  };
+
+  cargoSha256 = "0iv8zvglwaihcc89dk9kkhchbj1g3v8wq8jcbrgcbclcsyymmplc";
+
+  buildInputs = (stdenv.lib.optional stdenv.isDarwin Security);
```
Comment (marsam) [inline pkgs/tools/text/amber/default.nix]: nitpick, the parens aren't necessary here
```suggestion
  buildInputs = stdenv.lib.optional stdenv.isDarwin Security;
```

**Comment by marsam** [comment]
@GrahamcOfBorg build amber

**Comment by marsam** [comment]
Sorry for the delay. I just squashed the last commit, I hope you don't mind
Thanks @bdesham!

**Comment by bdesham** [comment]
@marsam Not at all! I prefer to squash/rebase whenever possible. Thank you!

**Pull request merged**